File: src/test/scala/com/exasol/cloudetl/bucket/HDFSBucketTest.scala
package com.exasol.cloudetl.bucket

class HDFSBucketTest extends AbstractBucketTest {

  test("apply sets correct configuration") {
    val properties = Map(
      PATH -> "hdfs://dir/path",
      FORMAT -> "orc"
    )
    val exaMetadata = mockConnectionInfo("", "")
    val bucket = getBucket(properties, exaMetadata)
    val expectedFileSystemName = classOf[org.apache.hadoop.hdfs.DistributedFileSystem].getName()
    assert(bucket.isInstanceOf[HDFSBucket])
    assert(bucket.getConfiguration().get("fs.hdfs.impl") === expectedFileSystemName)
  }

}

File: src/test/java/com/exasol/cloudetl/bucket/HDFSBucketTest.java
package com.exasol.cloudetl.bucket;

import static org.junit.jupiter.api.Assertions.*;

import java.util.Map;

import org.junit.jupiter.api.Test;

class HDFSBucketTest extends AbstractBucketTest {
    @Test
    void createSetsCorrectConfiguration() {
        final Map<String, String> properties = Map.of(PATH, "hdfs://dir/path", FORMAT, "orc");
        final Bucket bucket = getBucket(properties, mockConnectionInfo("", ""));
        assertInstanceOf(HDFSBucket.class, bucket);
        assertEquals(org.apache.hadoop.hdfs.DistributedFileSystem.class.getName(),
                bucket.getConfiguration().get("fs.hdfs.impl"));
    }
}
